### PR TITLE
Fix azure provider breaking change

### PR DESCRIPTION
**Pull request by Lee-W**
This PR fixes the failure due to https://pypi.org/project/apache-airflow-providers-microsoft-azure/8.0.0rc1/

### CODE REVIEW

File: astronomer/providers/microsoft/azure/example_dags/example_adf_run_pipeline.py [hunk 1]
```diff
@@ -97,17 +95,17 @@ def create_adf_storage_pipeline() -> None:
...
-        properties=AzureStorageLinkedService(connection_string=storage_string)
+        properties=AzureStorageLinkedService(connection_string=storage_string)  # type: ignore[arg-type]
     )
     adf_client.linked_services.create_or_update(
         RESOURCE_GROUP_NAME, DATAFACTORY_NAME, STORAGE_LINKED_SERVICE_NAME, ls_azure_storage
     )
 
     # Create an Azure blob dataset (input)
-    ds_ls = LinkedServiceReference(reference_name=STORAGE_LINKED_SERVICE_NAME)
+    ds_ls = LinkedServiceReference(type="LinkedServiceReference", reference_name=STORAGE_LINKED_SERVICE_NAME)
```
Comment (pankajkoti) [inline astronomer/providers/microsoft/azure/example_dags/example_adf_run_pipeline.py:105]: Do we need to support/take care of backward compatibility for the previous versions of the provider?

File: astronomer/providers/microsoft/azure/example_dags/example_adf_run_pipeline.py [hunk 2]
```diff
@@ -125,8 +123,8 @@ def create_adf_storage_pipeline() -> None:
     # Create a copy activity
     blob_source = BlobSource()
     blob_sink = BlobSink()
-    ds_in_ref = DatasetReference(reference_name=DATASET_INPUT_NAME)
-    ds_out_ref = DatasetReference(reference_name=DATASET_OUTPUT_NAME)
+    ds_in_ref = DatasetReference(type="DatasetReference", reference_name=DATASET_INPUT_NAME)
+    ds_out_ref = DatasetReference(type="DatasetReference", reference_name=DATASET_OUTPUT_NAME)
```
Comment (pankajkoti) [inline astronomer/providers/microsoft/azure/example_dags/example_adf_run_pipeline.py:127]: same question for backward compatibility.

**Comment by pankajkoti** [comment]
Looks like there is a mypy failure, also could we add a screenshot of a successful DAG run (either local or on Astro Cloud)?

**Comment by Lee-W** [comment]
> Looks like there is a mypy failure, also could we add a screenshot of a successful DAG run (either local or on Astro Cloud)?

I think it's because it was not released by then. Yep, let me deploy one more time. The DAG disappeared (as expected) . cannot screenshot the last successful DAG run

File: astronomer/providers/microsoft/azure/example_dags/example_adf_run_pipeline.py (hunk 1 above)
Comment (Lee-W) [reply astronomer/providers/microsoft/azure/example_dags/example_adf_run_pipeline.py:105]: And we used to install azure-mgmt-datafactory >= 1.0.0 here.

Comment (Lee-W) [reply astronomer/providers/microsoft/azure/example_dags/example_adf_run_pipeline.py:105]: Also as an example DAG, I think we should use the newer version. WDYT?

**Review by pankajastro — APPROVED** [review]
LGTM

**Pull request merged**